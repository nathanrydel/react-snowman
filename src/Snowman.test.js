import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";

const MAX_WRONG = 5

// maybe render out a Snowman instance in a beforeEach and test off that?

// SMOKE TEST
describe("basic rendering smoke test", function () {
  test("successful render", function () {
    render(<Snowman />)
  })
})


// TEST ONLY SIX WRONG GUESSES
// -- make sure some elements where not in the DOM -> make sure keyboard is disabled after meeting lose state (MAXWRONG GUESSES)

describe("Max failed guesses behavior", function () {
  // test winning condition
  // make a snap shot

  // test the losing condition
  // make a snapshot

  test("test hitting losing condition", function () {
    // fireEvents of wrong guess until hitting MAX_WRONG
    const {container, debug } = render(<Snowman maxWrong={ MAX_WRONG } words={["test"]} />)

    fireEvent.click(container.querySelector('button[value="a"]'))
    fireEvent.click(container.querySelector('button[value="b"]'))
    fireEvent.click(container.querySelector('button[value="c"]'))
    fireEvent.click(container.querySelector('button[value="d"]'))
    fireEvent.click(container.querySelector('button[value="f"]'))
    // Expect failure behavior
    // "YOU LOSE"
    // All buttons are disabled

    //TODO: figure out how to test these conditions properly
    expect(container).toHaveHTMLString("You lose: test") // FIXME:
    expect(container).toHaveHTMLString("Number wrong: 5") // FIXME:
    expect(container.querySelector('button[value="q"]')).not.toExist() //FIXME:
  })

  // SNAPSHOT -- holds the image of no buttons
})

// TEST each function in snowman