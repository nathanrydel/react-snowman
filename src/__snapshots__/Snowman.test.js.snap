// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Max failed guesses behavior matches losing snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="5"
      src="5.png"
    />
    <p>
      Number wrong: 
      5
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
    </p>
    <p>
      You lose: 
      test
    </p>
  </div>
</div>
`;
